10.0/Dockerfile

```diff
@@ -44,25 +44,40 @@ RUN set -x; \
 
 # grab gosu for easy step-down from root and dockerize to generate template and
 # wait on postgres
-RUN /install/gosu.sh && /install/dockerize.sh
+RUN /install/dockerize.sh
 
 COPY ./src_requirements.txt /odoo
-COPY ./bin /odoo-bin
-COPY ./bin-py2 /odoo-bin/
+COPY ./bin /odoo/odoo-bin
+COPY ./bin-py2 /odoo/odoo-bin/
 COPY ./templates /templates
 COPY ./before-migrate-entrypoint.d /before-migrate-entrypoint.d
 COPY ./start-entrypoint.d /start-entrypoint.d
 COPY ./MANIFEST.in /odoo
 # Place coveragerc in workdir where coverage will be launched from
 COPY ./.coveragerc /
 
-VOLUME ["/data/odoo", "/var/log/odoo"]
+RUN adduser --disabled-password -u 999 --gecos '' odoo
+RUN chown -R odoo:odoo /odoo
+RUN chown -R odoo:odoo /before-migrate-entrypoint.d
+RUN chown -R odoo:odoo /start-entrypoint.d
+
+RUN usermod odoo --home /odoo
+RUN mkdir -p /data/odoo/{addons,filestore,sessions} 
```

Comment:
create the directories before running the chown commands so you can set the owner on them together with the rest